### DISCUSSION TITLE
mTLS network encryption

### DISCUSSION THREAD

**Discussion by mferi87** [minio/minio#15697] [Q&A] [answered]
I was searching for some documentation about mTLS network encryption. In the documentation, I could find info about the TLS encryption (https://docs.min.io/minio/baremetal/security/network-encryption/minio-tls.html) but nothing about mTLS. Is it possible to use mTLS from config or we need a sidecar nginx for mTLS termination?

**Comment by ravindk89** [comment, marked as answer]
https://github.com/minio/minio/blob/master/docs/sts/tls.md <- I think this is as close as you can get to what you are looking for. It specifically uses an extension we built for the Security Token Service to allow for authentication using an x.509 certificate.

We are working on re-designing our documentation, and have somewhat paused on content to focus our efforts there (We're nearly done!)

Once that's finished, we'll be focusing on bringing everything up to parity, especially w.r.t. some of the power features we've built on the IAM side.

**Reply by mferi87** [reply]
Hi!
Many thanks for the prompt reply. It looks promising, we will try it out!
